tests/functional/aws-node-sdk/test/object/rangeTest.js

```diff
@@ -0,0 +1,254 @@
+        if (err) {
+            return cb(err);
+        }
+        const { begin, end } = getOuterRange(range, fileSize);
+        const contentLength = getTotal(begin, end).toString();
+        // If the range header is invalid, content range should be `undefined`
+        const contentRange = range === '-' ? undefined :
+            `bytes ${begin}-${end}/${fileSize}`;
+
+        assert.deepStrictEqual(res.ContentLength, contentLength);
+        assert.deepStrictEqual(res.ContentRange, contentRange);
+        assert.deepStrictEqual(res.ContentType, 'application/octet-stream');
+        assert.deepStrictEqual(res.Metadata, {});
+        assert.deepStrictEqual(res.Body, data);
+        return cb();
+    }))
+    .catch(cb);
+};
+
+// Complete the MPU, or abort it if there is an error
+const completeMPU = (range, fileSize, uploadId, cb) =>
+    s3.completeMultipartUploadAsync({
+        Bucket: bucket,
+        Key: key,
+        UploadId: uploadId,
+        MultipartUpload: {
+            Parts: [
+                {
+                    ETag: 'fa5a73131c61ee90e128fe6bef8544e2',
+                    PartNumber: 1,
+                },
+                {
+                    ETag: 'e09d29b0c1da7495508d56a4047fec71',
+                    PartNumber: 2,
+                },
+            ],
+        },
+    })
+    .then(() => checkRanges(range, fileSize, cb))
+    .catch(err => s3.abortMultipartUploadAsync({
+        Bucket: bucket,
+        Key: key,
+        UploadId: uploadId,
+    })
+    .then(() => cb(err)));
+
+// Create a hashed file of size n
+const createHashedFile = (fileSize, cb) => {
+    const name = `hashedFile.${fileSize}`;
+    execFile('gcc', ['-o', 'getRangeExec', 'lib/utility/getRange.c'],
+        err => {
+            if (err) {
+                return cb(err);
+            }
+            return execFile('./getRangeExec', ['--size', fileSize, name], cb);
+        });
+};
+
+// Use the hashed file to create a new file comprised of a byte range
+const createRangedFile = (range, fileSize, cb) => {
+    const name = `hashedFile.${fileSize}`;
+    const { begin, end } = getOuterRange(range, fileSize);
+    const total = getTotal(begin, end);
+    return execFile('dd', [`if=${name}`, `of=${name}.${range}`, 'bs=1',
+        `skip=${begin}`, `count=${total}`], err => cb(err, name));
+};
+
+// Create a file comprised of a byte range from the original hashed file
+const putRangeTest = (range, fileSize, cb) =>
+    createRangedFile(range, fileSize, (err, name) =>
+        // Add the original hashed file to the bucket for objectGet API
+        s3.putObjectAsync({
+            Bucket: bucket,
+            Key: key,
+            Body: fs.createReadStream(name),
+        })
+        .then(() => checkRanges(range, fileSize, cb))
+        .catch(cb));
+
+// Upload a hashed file as a MPU, then Create two 5MB parts to be uploaded as a
+// MPU, or abort it if there is an error
+const mpuRangeTest = (range, fileSize, uploadId, cb) =>
```

Comment:
I refactored the test so that it uses the executable of getRange.c to check the bytes of the object retrieved from the bucket instead of creating a new file with `dd` and comparing it. This allows for a cleaner `beforeEach` and `afterEach`. It also makes for a stronger test, I think, and makes the test closer to the sample.